docs/configs-api.md

```diff
@@ -0,0 +1,95 @@
+# Configs API
+
+The configs api provides API-driven multi-tenant approach to prometheus' Rule files and Alertmanager configs.
```

Comment:
minor point but I would add something like "backed by a database" - this doc mostly talks about files but the whole reason this code exists is because we wanted to use a database and the comparable code in Prometheus is hard-coded to only read files from a disk path.